subgraph/src/TroveNFT.mapping.ts

```diff
@@ -0,0 +1,68 @@
+import { Address, BigInt, dataSource } from "@graphprotocol/graph-ts";
+import { BorrowerInfo, Trove } from "../generated/schema";
+import { Transfer as TransferEvent } from "../generated/templates/TroveNFT/TroveNFT";
+
+enum BorrowerInfoUpdate {
+  add,
+  remove,
+}
+
+function updateBorrowerInfo(
+  borrowerAddress: Address,
+  troveFullId: string,
+  update: BorrowerInfoUpdate,
+): void {
+  let trove = Trove.load(troveFullId);
+  if (!trove) {
+    throw new Error("Trove does not exist: " + troveFullId);
+  }
+
+  let borrowerInfo = BorrowerInfo.load(borrowerAddress.toHexString());
+
+  if (!borrowerInfo && update == BorrowerInfoUpdate.add) {
+    borrowerInfo = new BorrowerInfo(borrowerAddress.toHexString());
+    borrowerInfo.troves = 0;
+
+    let totalCollaterals = dataSource.context().getI32("totalCollaterals");
+    borrowerInfo.trovesByCollateral = (new Array<i32>(totalCollaterals)).fill(0);
+  }
+
+  if (!borrowerInfo) {
+    throw new Error("BorrowerInfo does not exist: " + borrowerAddress.toHexString());
+  }
+
+  let diff = update == BorrowerInfoUpdate.add ? 1 : -1;
+
+  let trovesByColl = borrowerInfo.trovesByCollateral;
+  let collIndex = <i32> parseInt(trove.collateral);
+  trovesByColl[collIndex] += diff;
+
+  borrowerInfo.trovesByCollateral = trovesByColl;
+  borrowerInfo.troves += diff;
+  borrowerInfo.save();
+}
+
+export function handleTransfer(event: TransferEvent): void {
+  // Minting doesn’t need to be handled as we are already
+  // handling OP_OPEN_TROVE & OP_OPEN_TROVE_AND_JOIN_BATCH
+  // in TroveManager.mapping.ts.
+  if (event.params.from.equals(Address.zero())) {
+    return;
+  }
+
+  let collId = dataSource.context().getString("collId");
+  let troveFullId = collId + ":" + event.params.tokenId.toHexString();
+
+  // update BorrowerInfo for the previous owner
+  updateBorrowerInfo(event.params.from, troveFullId, BorrowerInfoUpdate.remove);
+
+  // update BorrowerInfo for the new owner (including zero address)
+  updateBorrowerInfo(event.params.to, troveFullId, BorrowerInfoUpdate.add);
```

Comment:
So burnt (closed) Troves will accumulate under the BorrowerInfo of address zero, is that right? Do we use this for something?

Reply:
No I just did this for consistency, so that all debt can be queried. It’s not used at the moment.

Reply:
But all Troves under address zero would have zero debt, no?

Reply:
My assumption is that the NFT of a trove could get transferred to the zero address without closing the trove. Is it incorrect?

Reply:
Ah, I missed this. To answer: no, you're not allowed to transfer to the zero address:
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/bd325d56b4c62c9c5c1aff048c37c6bb18ac0290/contracts/token/ERC721/ERC721.sol#L335

Reply:
Oh I didn’t know, thanks! I’ll add a check for `0x0` then :+1: